.github/workflows/ci.yml

```diff
@@ -0,0 +1,89 @@
+name: CI
+
+# Trigger the CI on pull requests and direct pushes to any branch
+on:
+  push:
+  pull_request:
+
+
+jobs:
+
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    # Pull requests from the same repository won't trigger this checks as they were already triggered by the push
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
+    steps:
+      - name: Clone the code
+        uses: actions/checkout@v2
+      - name: Execute golangci-lint
+        uses: golangci/golangci-lint-action@v2
+        with:
+          version: v1.29        # Always uses the latest patch version.
+          only-new-issues: true # Show only new issues if it's a pull request
+
+  testdata:
+    name: Verify testdata directory
+    needs: lint
+    runs-on: ubuntu-latest
+    # Pull requests from the same repository won't trigger this checks as they were already triggered by the push
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
+    steps:
+      - name: Clone the code
+        uses: actions/checkout@v2
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '^1.15'
+      # This step is needed as the following one tries to remove
+      # kustomize for each test but has no permission to do so
```

Comment:
It also tries to remove controller-gen iirc, is that not a problem?

Reply:
No, the problem is that the base virtual environment comes with `kustomize` installed under `/usr/local/bin` which requires `sudo` privileges to remove. The versions installed by us are fine but the first time the test is executed `rm -f $(command -v kustomize)` finds this one and can't remove it.